src/net_processing.cpp

```diff
@@ -3258,6 +3403,331 @@ bool static ProcessMessage(CNode* pfrom, const std::string& strCommand, CDataStr
         return true;
     }
 
+    // Received from an inbound peer planning to reconcilie transactions with us, or
+    // from an outgoing peer demonstrating readiness to do reconciliations.
+    // If received from outgoing, adds the peer to the reconciliation queue.
+    if (strCommand == NetMsgType::SENDRECON) {
+        if (!pfrom->m_tx_relay) return true;
+        if(pfrom->m_recon_state != nullptr) return true; // Do not support reconciliation salt/version updates.
+
+        bool recon_sender, recon_responder;
+        uint64_t remote_salt;
+        uint32_t recon_version;
+        vRecv >> recon_sender >> recon_responder >> recon_version >> remote_salt;
+        if (recon_version != 1) return true;
+
+        // According to current erlay spec.
+        if (recon_sender == recon_responder) return true;
+
+        if (pfrom->fInbound) {
+            if (!recon_sender) return true;
+            // Do not flood through inbound connections which support reconciliation to save bandwidth.
+            pfrom->m_flood_to = false;
+        } else {
+            if (!recon_responder) return true;
+            uint64_t outbound_flooding = connman->GetOutboundCountByTxRelayType(true);
+            if (outbound_flooding > MAX_OUTBOUND_FLOOD_TO)
+                pfrom->m_flood_to = false;
+        }
+
+        pfrom->m_recon_state = MakeUnique<CNode::ReconState>();
+        pfrom->m_recon_state->sender = recon_sender;
+        pfrom->m_recon_state->responder = recon_responder;
+        pfrom->m_recon_state->local_q = DEFAULT_RECON_Q;
+
+        uint64_t local_salt = connman->m_local_recon_salt;
+
+        std::string salt1, salt2;
+        if (remote_salt < local_salt) {
+            salt1 = std::to_string(remote_salt);
```

Comment:
I'm afraid `std::to_string` is locale dependent.

---

For those interested in permanently killing the locale dependency bug class, consider reviewing:
* #18124 – init: Clarify C and C++ locale assumptions. Add locale sanity checks to verify assumptions at run-time
* #18126 – tests: Add fuzzing harness testing the locale independence of the `strencodings.h` functions
* #18147 – qt: Kill the locale dependency bug class by not allowing Qt to mess with `LC_NUMERIC`